apps/web/src/features/events/components/events-map.tsx

```diff
@@ -324,69 +354,311 @@ export const EventsMap = ({
+                    strokeLinejoin="round"
+                    d="M10.5 6h9.75M10.5 6a1.5 1.5 0 1 1-3 0m3 0a1.5 1.5 0 1 0-3 0M3.75 6H7.5m3 12h9.75m-9.75 0a1.5 1.5 0 0 1-3 0m3 0a1.5 1.5 0 0 0-3 0m-3.75 0H7.5m9-6h3.75m-3.75 0a1.5 1.5 0 0 1-3 0m3 0a1.5 1.5 0 0 0-3 0m-9.75 0h9.75"
+                  />
+                </svg>
+                Filters
+              </div>
+            </div>
+
+            <div className="flex gap-4 font-light">
+              {courseTypes.map((f) => (
+                <button
+                  key={f}
+                  onClick={() => onFilterClick(f)}
+                  className={cn(
+                    'text-md border-b border-transparent capitalize',
+                    filter.includes(f)
+                      ? 'border-newOrange-1 font-semibold'
+                      : '',
+                  )}
+                >
+                  {f}s
+                </button>
+              ))}
+            </div>
+
+            <div>{/* Keep for spacing  */}</div>
+          </div>
+
+          <div
+            id="ol-map"
+            className={cn(
+              'w-full h-96 xl:h-[32rem] overflow-hidden',
+              !(selectedEventGroup || filter.length > 0) &&
+                (mode === DisplayMode.Calendar
+                  ? 'rounded-br-xl'
+                  : 'rounded-b-xl'),
+            )}
+          ></div>
+
+          {/* Switch mode */}
+          <div className="absolute bottom-2 left-2 hidden xl:block">
+            <Button
+              variant="newPrimary"
+              size="s"
+              className="h-8 border border-darkOrange-5 flex gap-2"
               onClick={() =>
-                setFilter((prev) =>
-                  prev.includes(f) ? prev.filter((p) => p !== f) : [...prev, f],
+                setMode(
+                  mode === DisplayMode.Calendar
+                    ? DisplayMode.Map
+                    : DisplayMode.Calendar,
                 )
               }
-              className={cn(
-                'text-md border-b border-transparent capitalize',
-                filter.includes(f) ? 'border-newOrange-1 font-semibold' : '',
-              )}
             >
-              {f}s
-            </button>
-          ))}
+              {mode === DisplayMode.Calendar ? (
+                <>
+                  <svg
+                    xmlns="http://www.w3.org/2000/svg"
+                    fill="none"
+                    viewBox="0 0 24 24"
+                    strokeWidth="1.5"
+                    stroke="currentColor"
+                    className="size-4"
+                  >
+                    <path
+                      strokeLinecap="round"
+                      strokeLinejoin="round"
+                      d="M15.75 19.5 8.25 12l7.5-7.5"
+                    />
+                  </svg>
+
+                  <span>Full map</span>
+                </>
+              ) : (
+                <>
+                  <svg
```

Comment:
Could we use react icons, or create like "arrow-left.svg" files instead of this ?